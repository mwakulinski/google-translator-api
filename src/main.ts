import { NestFactory } from '@nestjs/core';
import { AppModule } from './app.module';
import { setMainConfig } from './main.config';

async function bootstrap() {
  const app = await NestFactory.create(AppModule);
  setMainConfig(app);
  await app.listen(3000);
}
bootstrap();
