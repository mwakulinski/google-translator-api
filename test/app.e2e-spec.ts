import { Test, TestingModule } from '@nestjs/testing';
import { INestApplication } from '@nestjs/common';
import * as request from 'supertest';
import { AppModule } from './../src/app.module';
import { setMainConfig } from '../src/main.config';

describe('AppController (e2e)', () => {
  let app: INestApplication;

  beforeEach(async () => {
    const moduleFixture: TestingModule = await Test.createTestingModule({
      imports: [AppModule],
    }).compile();

    app = moduleFixture.createNestApplication();
    setMainConfig(app);
    await app.init();
  });

  describe('/translator [POST]', () => {
    it('properly sends request to endpoint', async () => {
      const { status } = await request(app.getHttpServer())
        .post('/translator')
        .send({ language: 'en' });

      expect(status).toBe(201);
    });

    it('throws error when inappropriate data send', async () => {
      const { body: response, status: status } = await request(
        app.getHttpServer(),
      )
        .post('/translator')
        .send({ language: 123 });

      expect(status).toBe(400);
      expect(response).toEqual({
        error: 'Bad Request',
        message: [
          'language must be longer than or equal to 2 characters',
          'language must be a string',
        ],
        statusCode: 400,
      });
    });
  });

  it('translates text to given language', async () => {
    const { text, status } = await request(app.getHttpServer())
      .post('/translator')
      .send({ language: 'en' });

    expect(status).toBe(201);
    expect(text).toMatchSnapshot();
  });

  afterAll(async () => {
    await app.close();
  });
});
