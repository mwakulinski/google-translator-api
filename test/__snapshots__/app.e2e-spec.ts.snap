// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppController (e2e) translates text to given language 1`] = `"{\\"attention\\":{\\"title\\":\\"Good that you are here, check this assignment\\",\\"subtitle\\":\\"It will help you figure out how to switch languages in React apps\\",\\"ctaButton\\":\\"Find out more\\",\\"array\\":[\\"Hello\\",\\"how\\",\\"myself\\",\\"you have\\",{\\"nested\\":\\"object\\"}]},\\"newsletter\\":{\\"title\\":\\"Good that you are here, check this assignment\\",\\"ctaButton\\":\\"It will help you figure out how to switch languages in React apps\\",\\"action\\":\\"Find out more\\",\\"object\\":{\\"it\\":\\"Hello\\"}}}"`;
